core/app/api/customer/groups/route.ts

```diff
@@ -0,0 +1,38 @@
+import { NextResponse } from 'next/server';
+import { z } from 'zod';
+
+export const CustomersGroup = z
+  .object({
+    id: z.number(),
+    name: z.string(),
+  })
+  .array();
+
+export type CustomerGroupType = z.infer<typeof CustomersGroup>;
+
+export async function GET() {
+  if (!process.env.BIGCOMMERCE_ACCESS_TOKEN) {
+    throw new Error('BIGCOMMERCE_ACCESS_TOKEN is not set');
+  }
+
+  const response = await fetch(
+    `https://api.bigcommerce.com/stores/${process.env.BIGCOMMERCE_STORE_HASH}/v2/customer_groups`,
+    {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        'X-Auth-Token': process.env.BIGCOMMERCE_ACCESS_TOKEN,
+      },
+    },
+  );
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  const jsonResponse: unknown = await response.json();
+  const groups = CustomersGroup.parse(jsonResponse);
```

Comment:
This can still throw an error. Alternatively, you can use `safeParse()` which will return an error but not throw one. I would wrap this whole function  in try/catch. Then, for the error, you can do checking on the type of the error to return specific messages. In this case, you can check for instance of `ZodError`.

You can see an example in the Zod docs here -  https://zod.dev/ERROR_HANDLING?id=a-demonstrative-example

Reply:
So you would check in your top-level try/catch instanced of a zod specific error as well as a generic error.